Sources/BraveNews/Composer/FeedDataSource.swift

```diff
@@ -775,29 +776,32 @@ public class FeedDataSource: ObservableObject {
   ) {
     // Ensure main thread since we're querying from CoreData
     dispatchPrecondition(condition: .onQueue(.main))
-    let lastVisitedDomains =
-      (try? History.suffix(200)
-        .lazy
-        .compactMap(\.url)
-        .compactMap { URL(string: $0)?.baseDomain }) ?? []
-    let followedSources = FeedSourceOverride.all().filter(\.enabled).map(\.publisherID)
-    todayQueue.async {
-      let items: [FeedItem] = feeds.compactMap { content in
-        var score = content.baseScore ?? Double.greatestFiniteMagnitude
-        if let feedBaseDomain = content.url?.baseDomain,
-          lastVisitedDomains.contains(feedBaseDomain) {
-          score -= 5
-        }
-        guard let source = sources.first(where: { $0.id == content.publisherID }) else {
-          return nil
+    
+    var lastVisitedDomains: [String] = []
+
+    historyAPI?.search(withQuery: "", maxCount: 200) { historyNodeList in
```

Comment:
Since history is not checked before calling, this will cause the `score` to never call the completion handler if it is  nil. We should add a guard against it or ensure history API is non-nil before scoring

Reply:
Fix at https://github.com/brave/brave-ios/pull/7087/commits/e5c8750c082fb288e8b9c45c1c87a1c810c84b98